slidingwindow.go

```diff
@@ -147,3 +151,87 @@ func (s *SlidingWindowRedis) Increment(ctx context.Context, prev, curr time.Time
 		return 0, 0, ctx.Err()
 	}
 }
+
+// SlidingWindowDynamoDB implements SlidingWindow in DynamoDB.
+type SlidingWindowDynamoDB struct {
+	fixedWindow *FixedWindowDynamoDB
```

Comment:
I think it will be clearer if the `FixedWindowDynamoDB` is not reused here. Instead add sliding window specific fields and methods.